src/containers/forms/CreateCategoryForm.jsx

```diff
@@ -0,0 +1,189 @@
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import SnackBar from '../../components/notifications/SnackBar';
+import SingleInput from '../../common/SingleInput';
+import Button from '../../common/Button';
+import TextArea from '../../common/TextArea';
+import FormError from '../../components/formErrors/FormError';
+import { createCategory } from '../../actions/createCategoryActions';
+import validateFormFields from '../../helpers/validate';
+import capitalizeString from '../../helpers/stringFormatter';
+
+/**
+   * @name CreateCategoryForm
+   * @summary Returns Form
+   * @returns Returns a form
+   */
+class CreateCategoryForm extends Component {
+  /**
+   * @name propTypes
+   * @type {PropType}
+  */
+  static propTypes = {
+    closeModal: PropTypes.func.isRequired,
+    createCategory: PropTypes.func.isRequired,
+  };
+
+  /**
+   * CreateCategoryForm component class constructor
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: '',
+      name: '',
+      description: '',
+      errors: [],
+      message: null,
+    };
+  }
+
+  static getDerivedStateFromProps = (nextProps) => {
+    // clear form fields if category was created successfully
+    if (nextProps.message && nextProps.message.type === 'success') {
+      return {
+        value: '',
+        name: '',
+        description: '',
+        errors: [],
+        message: nextProps.message,
+      };
+    }
+    return {
+      message: nextProps.message,
+    };
+  };
+  /**
+   * @memberOf CreateCategoryForm
+   * change event handler
+   * @param {Event} event - change event on select input
+   */
+  handleChange = (event) => {
+    this.setState({ [event.target.name]: event.target.value });
+    // if input value is not empty remove it from error list
+    if (!this.state[event.target.name]) {
+      const errors = this.state.errors.filter(error => !error.includes(event.target.name));
+      this.setState({ errors });
+    }
+  }
+
+  handleAddEvent = (event) => {
+    event.preventDefault();
+    const { name, description, value } = this.state;
+    const category = {
+      name,
+      description,
+      value,
+    };
+    this.setState({
+      errors: validateFormFields(category),
+    }, () => {
+      if (this.state.errors.length === 0) {
+        this.props.createCategory(category);
+      }
+    });
+  }
+
+  /**
+   * @name resetState
+   * @summary resets state to clear form fields and error messages
+   */
+  resetState = () => {
+    this.setState({
+      name: '',
+      value: '',
+      description: '',
+      errors: [],
+      message: null,
+    });
+  }
+
+  /**
+   * @name cancelModal
+   * @summary reset state and close modal
+   */
+  cancelModal = (event) => {
+    event.preventDefault();
+    this.resetState();
+    this.props.closeModal();
+  }
+
+  renderValidationError = (field, replaceWord) => {
+    if (this.state.errors.indexOf(field) >= 0) {
```

Comment:
Briefly, what does this do?

Reply:
This renders the error message in the specified area in the modal when there is a validation error in the data to be submitted

Reply:
Aha. So the fields that have errors are maintained if they exist somewhere and then evaluated using this function

Reply:
Yes please. So everywhere the function is used with an input to be validated so that it can render the errors for it in the right location